File: Daniel Yip/week10/burning-airlines-immersive/ba-node-backend/models/seeds.js
const mongoose = require('mongoose');

//Load our Flight model file (and any others?)
const Flight = require('./Flight');
const User = require('./User');


// Connect to DB server

mongoose.connect('mongodb://127.0.0.1/ba')

const db = mongoose.connection;

db.on('error', (err)=>{
  console.log('Connection error', err);
});

// Once the connection is established, we can start querying/seeding

db.once('open', async () => {

  await User.deleteMany({});

  let u1Id
  let u2Id

  try {
    const users = await User.create([
      {
        name: 'Test User 1',
        email: 'one@one.com'
      },
      {
        name: 'Test User 2',
        email: 'two@two.com'
      },
      {
        name: 'Test User 3',
        email: 'three@three.com'
      }
    ])
    u1Id = users[0]._id
    u2Id = users[1]._id
    console.log('USER 1 ID', u1Id);
    
  } catch (err) {
    console.log('error creating users', err);
  }

  await Flight.deleteMany({});

  try {
    await Flight.create([
      {
        flight_number: 'BA123',
        origin: 'SYD',
        destination: 'MEL',
        departure_date: new Date('2022-03-20T04:20:00Z'),
        airplane: {name: '737 MAX', rows: 20, cols: 6}, //nested data instead of an ID assocication
        reservations: [
          {row: 1, col: 1, user: u1Id},
          {row: 2, col: 2, user: u1Id},
          {row: 3, col: 3, user: u2Id}
        ],
      },
      {
        flight_number: 'BA456',
        origin: 'SYD',
        destination: 'MEL',
        departure_date: new Date('2022-03-21T04:20:00Z'),
        airplane: {name: '767', rows: 16, cols: 4}, //nested data instead of an ID assocication
        reservations: [
          {row: 1, col: 1, user: u1Id},
          {row: 1, col: 2, user: u1Id},
          {row: 1, col: 3, user: u2Id}
        ],
      }
    ])
    
  } catch (err) {
    console.log('error creating flights', err);
  }
  
  // console.log('Created flights:', results);
  
  try {
    // ActiveRecord Flight.all, but it returns a promise...
    const flights = await Flight.find()
    console.log(flights);
  } catch (error) {
    console.log('Error querying results', err);
    process.exit(1);
  }
  

  process.exit(0); // all good exit program

});

File: David Broomhead/wk10-friday/burning-airlines-immersive/ba-rails-backend/db/seeds.rb
User.destroy_all
print "Creating users...."

u1 = User.create! name: 'Test User 1', email: 'one@one.com', password: 'chicken'
u2 = User.create! name: 'Test User 2', email: 'two@one.com', password: 'chicken'
u3 = User.create! name: 'Test User 3', email: 'three@one.com', password: 'chicken'

puts "created #{User.count} users."

################################################################################################################################

Airplane.destroy_all
print "Creating airplanes....."

a1 = Airplane.create! name: '747', rows: 50, cols: 6
a2 = Airplane.create! name: '787', rows: 40, cols: 6
a3 = Airplane.create! name: '737', rows: 30, cols: 6

puts "created #{Airplane.count} airplanes."

################################################################################################################################

Flight.destroy_all
print "Creating flights..."

f1 = Flight.create! flight_number: 'BA256', departure_date: '2022-03-01 04:20', origin: 'SYD', destination: 'MEL', airplane: a1
f2 = Flight.create! flight_number: 'BA512', departure_date: '2022-03-02 05:20', origin: 'SYD', destination: 'ADL', airplane: a1
f3 = Flight.create! flight_number: 'BA1024', departure_date: '2022-03-03 06:20', origin: 'SYD', destination: 'SIN', airplane: a2

puts "created #{Flight.count} flights."

################################################################################################################################

Reservation.destroy_all
print "Creating reservations....."

r1 = Reservation.create! flight_id: f1.id, user_id: u1.id, row: 10, col: 2
r2 = Reservation.create! flight: f1, user: u2, row: 10, col: 3
r3 = Reservation.create! flight: f1, user: u3, row: 10, col: 4
r4 = Reservation.create! flight: f2, user: u3, row: 9, col: 4

puts "created #{Reservation.count} reservations."

puts "----------"
puts "Association tests:"
puts "Airplane 1 has #{Airplane.first.flights.length} flights (should be 2)."
raise Exception.new('Airplane flights association is broken') if Airplane.first.flights.length != 2

puts "Flight 1 has #{Flight.first.reservations.length} reservations (should be 3)."
puts "User 3 has #{User.third.reservations.length} reservations (should be 2."

puts "Flight 1 has airplane name #{Flight.first.airplane.name} (should be 747)."

puts "Reservation 1 has user name #{Reservation.first.user.name} (should be Test User 1)."

puts "Reservation 2 has flight name #{Reservation.second.flight.flight_number} (should be BA256)."

